### PR TITLE
feat: Enhance block import log line

### LINKED ISSUE #5856: Increase information density at INFO level for logs during steady state
### Describe the feature

When Reth has reached "steady state" - where it is validating incoming blocks - the logging produced is verbose and lacks data often found in other ethereum clients.  Currently it produces about 4 lines per block with minimal information and lots of boilerplate.  Logging should be reduced to one line per block processed at info and should maximize data available.

Logging info from the CL could be limited to when CL values change, such as what the justified/finalized blocks are at epoch.  Also logging when those are expected to change bug didn't may be informative but are more of a CL concern.  A simple mod won't work because of missed/forked blocks.

It may also be valuable to include extra info for when reth is asked to produce a block and not just validate a block. At the very least a indication in the log that this was reth produced instead of validated, but other info such as fees retained

### Additional context

Here is the current reth log for processing one block.  Five lines is too many.
```
2023-12-22T19:56:00.310347Z  INFO reth::node::events: Canonical chain committed number=18843440 hash=0x76ef35d4fe74aaa26533a9839d2c264daff538ec6916439924e498f941672542 elapsed=6.351069791s
2023-12-22T19:56:00.310355Z  INFO reth::node::events: Forkchoice updated head_block_hash=0x76ef35d4fe74aaa26533a9839d2c264daff538ec6916439924e498f941672542 safe_block_hash=0xc5b8cdf6a9e233c6eac41a13d8afe5d34b39d1ad1a13721467945289c5473edc finalized_block_hash=0xb8d70aa8ad02345d58b9fda41d116aef95d85b6a37c224bdab40f63ef1cd5aaa status=Valid
2023-12-22T19:56:01.984020Z  INFO reth::cli: Status connected_peers=66 freelist=3281170 latest_block=18843440
2023-12-22T19:56:03.044291Z  INFO reth::node::events: Block added to canonical chain number=18843441 hash=0x8b55150aa134f244558bafe2194b5cc061a0a2ab109e334d8cf36a4ee1f3384e
2023-12-22T19:56:10.832966Z  INFO reth::node::events: Canonical chain committed number=18843441 hash=0x8b55150aa134f244558bafe2194b5cc061a0a2ab109e334d8cf36a4ee1f3384e elapsed=7.75455275s
```

Logging at info should establish a set of UX principals (yes, log lines are UX).  The following are suggestions and can be dealt with as the team sees fit.

Rule 0 - If something is going wrong or user intervention is required, log whatever is needed.
Rule 1 - for processing a block output one line at the end of processing.
Rule 2 - Include transient and block related information as part of this log line
Rule 3 - Be concise.  Use abbreviations and punctuation

Possible info to include:
* Block hash, full or elided
* number of transactions
* number of blobs
* number of peers
* gas used and gas limit (maybe just as a % of the limit)
* gas base fee
* blob base fee 
* block height
* Time spent processing (ideally time from receipt from engine until response to CL that it is valid/invalid, but could include all reth tasks)
* In-memory transaction pool size

It may be worth it to ask some devops folks who have experience running other node software about their wishes.  They tend to have stronger opinions about stuff like structured logging, date time formats, and if hashes should be elided.

### PR DESCRIPTION
**Pull request by shemnon**
Adds gas, gas used, tx count, base fee, and import elapsed time to the block imported log line.

Resolves #5856

### CODE REVIEW

**Comment by shemnon** [comment]
sample line
```
2024-01-16T13:24:16.664029Z  INFO reth::commands::node::events: Block added to canonical chain	txs=222 mgas=14.450 gas_pct=48.17% base_fee_gwei=25.51 elapsed=5.22s number=19018271 hash=0x5f8b453ea9797c810e66e0c63642739ed54f792ac2248f1be502be1201334caf num_connected_peers=93
```

File: crates/consensus/beacon/src/engine/mod.rs [hunk 1]
```diff
@@ -1257,15 +1258,18 @@ where
         debug_assert!(self.sync.is_pipeline_idle(), "pipeline must be idle");
 
         let block_hash = block.hash;
+        let sw = Stopwatch::start_new();
         let status = self
             .blockchain
             .insert_block_without_senders(block.clone(), BlockValidationKind::Exhaustive)?;
+        let elapsed = sw.elapsed();
```
Comment (Rjected) [inline crates/consensus/beacon/src/engine/mod.rs:1265]: We can probably just use the std lib time primitives here rather than importing a new crate, for example using:
```rust
let start = Instant::now();
// ... thing you want to time
let elapsed = start.elapsed();
```

**Review by Rjected — DISMISSED** [review]

File: crates/consensus/beacon/src/engine/event.rs [hunk 2]
```diff
@@ -9,7 +9,7 @@ pub enum BeaconConsensusEngineEvent {
     /// The fork choice state was updated.
     ForkchoiceUpdated(ForkchoiceState, ForkchoiceStatus),
     /// A block was added to the canonical chain.
-    CanonicalBlockAdded(Arc<SealedBlock>),
+    CanonicalBlockAdded(Arc<SealedBlock>, Duration),
```
Comment (mattsse) [inline crates/consensus/beacon/src/engine/event.rs:12]: I'm okay with this but this should be documented what the duration is

File: bin/reth/src/commands/node/events.rs [hunk 3]
```diff
@@ -357,13 +385,19 @@ where
...
-                    connected_peers = this.state.num_connected_peers(),
-                    %freelist,
-                    %latest_block,
-                    "Status"
-                );
+                let now =
+                    SystemTime::now().duration_since(UNIX_EPOCH).unwrap_or_default().as_secs();
+                if now - this.state.latest_block_time.unwrap_or(0) > 60 {
+                    // Once we start receiving consensus nodes, don't emit status unless stalled for
+                    // 1 minute
```
Comment (mattsse) [inline bin/reth/src/commands/node/events.rs:392]: does this effectively disable this info log in normal operations?

do we want this and why?

File: bin/reth/src/commands/node/events.rs [hunk 4]
```diff
@@ -38,11 +37,20 @@ struct NodeState<DB> {
...
     /// The latest block reached by either pipeline or consensus engine.
     latest_block: Option<BlockNumber>,
+    /// The time of the latest block seen by the pipeline
+    latest_block_time: Option<u64>,
+    /// Hash of the head block last set by fork choice update
+    head_block_hash: B256,
+    /// Hash of the safe block last set by fork choice update
+    safe_block_hash: B256,
+    /// Hash of finalized block last set by fork choice update
+    finalized_block_hash: B256,
```
Comment (mattsse) [inline bin/reth/src/commands/node/events.rs:47]: are we okay with them being zero on launch?

File: crates/primitives/src/constants/mod.rs [hunk 5]
```diff
@@ -121,6 +121,9 @@ pub const ETH_TO_WEI: u128 = FINNEY_TO_WEI * 1000;
 /// Multiplier for converting mgas to gas.
 pub const MGAS_TO_GAS: u64 = 1_000_000u64;
 
+/// Multiplier for blob size
+pub const BLOB_SIZE_BYTES: u64 = 128 * 1024;
```
Comment (mattsse) [inline crates/primitives/src/constants/mod.rs:125]: unclear where this is coming from

**Review by mattsse — CHANGES_REQUESTED** [review]
ptal @Rjected

File: bin/reth/src/commands/node/events.rs (hunk 3 above)
Comment (shemnon) [reply bin/reth/src/commands/node/events.rs:392]: We wind up saying the same thing 3 different ways in the log.  once when imported, once when canoncial chain is updated, and once every time the status tick fires.  The info is already communicated via the other two means, including peer count.  The goal of the bug is to increase density of the logs and this is duplicate data.

Adding the time check brings the status back if blocks are behind.

File: crates/primitives/src/constants/mod.rs (hunk 5 above)
Comment (shemnon) [reply crates/primitives/src/constants/mod.rs:125]: From the EIP-4844 spec.  Relocated next to 1558 constants and renamed to match the spec.

File: bin/reth/src/commands/node/events.rs (hunk 4 above)
Comment (shemnon) [reply bin/reth/src/commands/node/events.rs:47]: Changed to `Option<B256>`

File: crates/consensus/beacon/src/engine/event.rs (hunk 2 above)
Comment (shemnon) [reply crates/consensus/beacon/src/engine/event.rs:12]: Documented

File: bin/reth/src/commands/node/events.rs (hunk 3 above)
Comment (mattsse) [reply bin/reth/src/commands/node/events.rs:403]: sounds reasonable, ty

File: crates/primitives/src/constants/mod.rs [hunk 6]
```diff
@@ -61,6 +61,9 @@ pub const EIP1559_DEFAULT_BASE_FEE_MAX_CHANGE_DENOMINATOR: u64 = 8;
 /// Elasticity multiplier as defined in [EIP-1559](https://eips.ethereum.org/EIPS/eip-1559)
 pub const EIP1559_DEFAULT_ELASTICITY_MULTIPLIER: u64 = 2;
 
+/// Size of blobs in bytes as defined in [EIP-4844](https://eips.ethereum.org/EIPS/eip-4844)
+pub const EIP4844_GAS_PER_BLOB: u64 = 2u64.pow(17);
```
Comment (mattsse) [inline crates/primitives/src/constants/mod.rs:65]: ah we have this already:

https://github.com/paradigmxyz/reth/blob/838e6179ffc1acf075ab761e6736adcabaad0708/crates/primitives/src/constants/eip4844.rs#L12-L12

**Review by mattsse — CHANGES_REQUESTED** [review]
last nit, 
pending @Rjected @shekhirin

File: crates/primitives/src/constants/mod.rs (hunk 6 above)
Comment (shemnon) [reply crates/primitives/src/constants/mod.rs]: Constant updated.

**Comment by shemnon** [comment]
Current log status line:

```
024-02-08T20:49:50.431152Z  INFO reth::commands::node::events: Block added to canonical chain number=904063 hash=0x9abdc5fae5bad6434d7e5f3da10ce66e0d9e95fe35afce807482e091a41f2af3 peers=7 txs=28 mgas=8.368 full=27.9% base_fee=1.05gwei blobs=6 excess_blobs=642 elapsed=91.610542ms
```

**Review by mattsse — APPROVED** [review]
ty!

lgtm, pending @shekhirin and @Rjected 

and yeah @shekhirin we should shorten the `reth::commands::node::events`

**Review by shekhirin — APPROVED** [review]
LGTM, nice. I like the `Forkchoice updated` logging condition change.

File: bin/reth/src/commands/node/events.rs [hunk 7]
```diff
@@ -155,19 +173,40 @@ impl<DB> NodeState<DB> {
...
+            BeaconConsensusEngineEvent::CanonicalBlockAdded(block, elapsed) => {
                 info!(
-                    ?head_block_hash,
-                    ?safe_block_hash,
-                    ?finalized_block_hash,
-                    ?status,
-                    "Forkchoice updated"
+                    number=block.number,
+                    hash=?block.hash(),
+                    peers=self.num_connected_peers(),
```
Comment (shekhirin) [inline bin/reth/src/commands/node/events.rs:196]: nit: should we display peers here? they're already in the `Status` message, and here it doesn't seem to be related to the block import

Comment (shemnon) [reply bin/reth/src/commands/node/events.rs:196]: Besu lists it, and that was something Afri added years ago.  I think parity listed it too in their lines.

As for status messages, if we are within a minute of sync they are suppressed because it's duplicate data (blocks come faster than the tick) - https://github.com/paradigmxyz/reth/pull/6095#discussion_r1481776119 - and peers was the only unique item. So it serves to keep the line count lower.

File: bin/reth/src/commands/node/events.rs [hunk 8]
```diff
@@ -357,13 +396,19 @@ where
...
-                info!(
-                    target: "reth::cli",
-                    connected_peers = this.state.num_connected_peers(),
-                    %freelist,
-                    %latest_block,
-                    "Status"
-                );
+                let now =
+                    SystemTime::now().duration_since(UNIX_EPOCH).unwrap_or_default().as_secs();
+                if now - this.state.latest_block_time.unwrap_or(0) > 60 {
```
Comment (shekhirin) [inline bin/reth/src/commands/node/events.rs:401]: I believe this can be rewritten as
```suggestion
                if this.state.latest_block_time.map_or(true, |latest_block_time| latest_block_time.elapsed() > Duration::from_secs(60) {
```

if we also change the type of `this.state.latest_block_time` to `Option<Instant>`

File: bin/reth/src/commands/node/events.rs (hunk 7 above)
Comment (shekhirin) [reply bin/reth/src/commands/node/events.rs:196]: ah, I see, right. Hmm, I just thought that it might be a bit confusing for users to see the number of peers alongside the block import -related data, as if peers are actually related to block import.

File: bin/reth/src/commands/node/events.rs (hunk 8 above)
Comment (shekhirin) [reply bin/reth/src/commands/node/events.rs:401]: ah, we set the `this.state.latest_block_time` from `head.timestamp` which is already `u64`. Then it's fine with me 👍

**Review by shekhirin — APPROVED** [review]
